### ISSUE TITLE
Every `source .zshrc` add the same values (repeat them) into $fpath

### ISSUE THREAD

**Issue by kreedz** [tarjoilija/zgen#89]
I've noticed that ${fpath} starts to repeat (equal to the number of sourcing) the paths that mentioned in the ~/.zgen/init.zsh. It's normal?
~/.zgen/init.zsh.:
```
# ### Plugins & Completions
fpath=(/home/kreedz/.zgen/zsh-users/zsh-autosuggestions-develop / ${fpath})
```
`.zshrc`:
```
source ~/.zgen/zgen.zsh
if ! zgen saved; then
    zgen load zsh-users/zsh-autosuggestions zsh-autosuggestions.zsh develop
    zgen save
fi
```
So if I do `source .zshrc` thrice than the $fpath will contain next paths (repeated three times):
/home/kreedz/.zgen/zsh-users/zsh-autosuggestions-develop
/home/kreedz/.zgen

Doing `zgen reset` before every `source` prevents that.

**Comment by Tuurlijk** [comment]
Yes, this is normal. But it will not occur when using zsh in a regular fashion.

What is your use-case for sourcing .zshrc multiple times?
